### DISCUSSION TITLE
Text input of multiple parts

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#5282] [Feature requests]
Can we have something like this out of the box for Text input?
The `localhost:3000` part is a part of the Text input.

<img width="518" alt="Screenshot 2023-11-18 at 11 13 02" src="https://github.com/mantinedev/mantine/assets/22228431/7e088885-5b84-45e0-b0c3-88e1891b235f">

**Comment by rtivital** [comment]
You can add right section to all inputs with `rightSection` prop.

**Reply by atumas-bananamilk** [reply]
Solution (lots of custom code):
```
<TextInput
  placeholder='Email address'
  label='Email address'
  className='w-[320px]'
  classNames={{
    section: 'overflow-hidden'
  }}
  rightSectionWidth={140}
  rightSection={(
    <div className="flex justify-center items-center h-full w-full bg-gray-0">
      localhost:3000
    </div>
  )}
/>
```
UI:
![Screenshot 2024-12-11 at 20 10 03](https://github.com/user-attachments/assets/7e9bea42-45f5-42b5-9d92-997b41270fd9)
